buildchain/buildchain/targets/remote_image.py

```diff
@@ -90,17 +90,34 @@ def _show(task: types.Task) -> str:
         return utils.title_with_target1('IMG PULL', task)
 
     def _build_actions(self) -> List[types.Action]:
+        """Compute actions for the image — pull, tag, save."""
         filepath = self.uncompressed_filename
+        pull_command = DockerPull(
+            repository=self.repository, digest=self.digest
+        )
+        # The local repository is the image 'tag' without version
+        local_repository = self.tag[self.tag.rfind(':')]
```

Comment:
This won't work: `a = "toto:1.2.3"; a[a.rfind(':')]` will return `":"`.

```suggestion
        local_repository = self.tag[:self.tag.rfind(':')]
```

However, I find this method of retrieving the repository quite obscure. If the goal is to have a remote repository `{self.registry}/{self._remote_name}` vs. a local repository `{self.registry}/{self.name}`, why not be explicit about it and have dedicated properties for both?

Reply:
Should be `self.tag[self.tag.rfind(':'):]` (note the second colon).

Fixing the names so we can be explicit and accurate about everything is #947